packages/next/src/server/future/route-modules/pages/builtin/_error.tsx

```diff
@@ -19,5 +19,6 @@ export const routeModule = new PagesRouteModule({
     App,
     Document,
   },
+  // @ts-expect-error -- Types don't account for getInitialProps. `Error` requires to be instantiated with `statusCode` but the types currently don't guarantee that.
   userland: moduleError,
```

Comment:
I suspect this previously worked due to class components having legacy context as the second arg in the constructor and then it somehow  matched the component type even though the props mismatched. Ideally we'd have a type that would automatically resolve the `getInitialProps` like we did for `defaultProps` so that we could explain that we need a component that takes no props except the ones it returns from `getInitialProps`.